lib/auth.js

```diff
@@ -0,0 +1,48 @@
+"use strict";
+
+var P       = require('bluebird');
+var rbUtil  = require('./rbUtil');
+var URI     = require('swagger-router').URI;
+
+var auth = {};
+
+auth.checkPermissions = function(restbase, req, permissions) {
+    var checkReq = {
+        uri: new URI([req.params.domain,'sys','action','userinfo']),
+        method: 'post',
+        body: {
+            meta: 'userinfo',
+            uiprop: 'rights'
+        }
+    };
+    rbUtil.copyForwardedHeaders(restbase, checkReq);
+    return restbase.post(checkReq)
+    .then(function(userInfo) {
+        var accessAllowed;
+        var absentPermissions = [];
+        if (userInfo.rights && Array.isArray(userInfo.rights)) {
+            permissions.forEach(function(perm) {
+                if (userInfo.rights.indexOf(perm) < 0) {
+                    absentPermissions.push(perm);
```

Comment:
You can error out immediately here and report just the first permission, no need to be thorough when access is denied.